flexget/components/sites/sites/serienjunkies.py

```diff
@@ -107,43 +179,82 @@ def parse_downloads(self, series_url, search_title):
             if not episode_title:
                 continue
 
-            # find download container
-            episode = episode_title.parent
-            if not episode:
-                continue
+            urls = self.find_urls(urls, episode_title)
+
+        if len(urls) == 0 and season_pack_fallback:
+            episode_match_pattern = re.compile('\.S(\d*)E(\d*)\.')
+            season_title_match = re.search(episode_match_pattern, search_title)
+            if season_title_match:
+                season_title = re.sub(episode_match_pattern, r'.S\1.', search_title)
+                season_titles = self.find_all_titles(season_title)
+                if not season_titles:
+                    raise UrlRewritingError('Unable to find season')
+
+                for se_title in season_titles:
+                    # find matching download
+                    season_title = soup.find('strong', text=re.compile(se_title, re.I))
+                    if not season_title:
+                        continue
+
+                    urls = self.find_urls(urls, season_title)
 
+        return urls
+
+    def find_urls(self, urls, episode_title):
+        # find download container
+        episode = episode_title.parent
+        do_exit = False
+        if not episode:
+            do_exit = True
+
+        if not do_exit:
             # find episode language
             episode_lang = episode.find_previous('strong', text=re.compile('Sprache')).next_sibling
             if not episode_lang:
-                logger.warning('No language found for: {}', series_url)
-                continue
+                logger.warning('No language found for: {}', episode_lang)
+                do_exit = True
```

Comment:
The way you use this flag is kind of awkward and makes your code really hard to follow. Can you refactor it to be cleaner?